src/libcore/char.rs

```diff
@@ -420,7 +430,8 @@ pub fn encode_utf16_raw(mut ch: u32, dst: &mut [u16]) -> Option<usize> {
 #[stable(feature = "rust1", since = "1.0.0")]
 pub struct EscapeUnicode {
     c: char,
-    state: EscapeUnicodeState
+    state: EscapeUnicodeState,
+    offset: usize,
```

Comment:
Similarly here? At the very least I’d like to have a comment as to what sort of `offset` this is (e.g. a comment).